### ISSUE TITLE
SA1011 (ClosingSquareBracketsMustBeSpacedCorrectly) false positive

### ISSUE THREAD

**Issue by Przemyslaw-W** [DotNetAnalyzers/StyleCopAnalyzers#2564]
I believe no SA1011 should be reported in following code:

```csharp
        public unsafe void Unsafe(Point pt)
        {
            fixed (int* p = &pt.X)
            {
                int*[] xx = new[] { p };
                Console.WriteLine(xx[0]->ToString("n2"));
            }
        }
```
